pkg/config/remote/uptane/client.go

```diff
@@ -0,0 +1,216 @@
+package uptane
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/pkg/errors"
+	"github.com/theupdateframework/go-tuf/client"
+	"github.com/theupdateframework/go-tuf/data"
+	"go.etcd.io/bbolt"
+)
+
+type State struct {
+	ConfigRootVersion     uint64
+	ConfigSnapshotVersion uint64
+	DirectorRootVersion   uint64
+}
+
```

Comment:
Some of the comments in the PR could be added to the code to describe it

Reply:
Added comments to most components, do you see anything missing now ?